### ISSUE TITLE
Klick'r thinks it's been denied permission, when it has not been denied.

### ISSUE THREAD

**Issue by SteegHobland** [Nain57/Smart-AutoClicker#507]
### Version of Smart AutoClicker

3.0.2

### Device brand and model

TCL REVVL 4

### Version of your Android OS

10

### Type

Other

### Describe what happened?

There are times that Klick'r stops running the scenario I've given it. Not that big a deal, but it does illogical things when I try to start it up again.

After I hit the app icon on my home screen, then hit play on the scenario I want to run, it requests accessibility permission. I hit the request permission button. In the accessibility permission screen, it lists Klick'r as on.

That's peculiar.

After backing out of that, I click outside of the request prompt to dismiss it (not hitting deny, and I haven't denied anything!).

Klick'r then says its permission denied message about how the permission is mandatory for it to work. That's the illogical part. It has not been denied, and it still has permission.

If I instead try turning the permission off in the accessibility screen, and then go back in to turn it on again, there's no difference. Klick'r still behaves as though denied.

The only way I can get it to behave somewhat normally again is to force stop the app, and start it again. Then when it requests permission, it actually reads as off, and then correctly acknowledges it has permission, after I grant it and exit the accessibility screen.

### Describe what should have happened?

It should behave as though it already has permission, and skip the request, when it already has permission. It should not needlessly request a permission it already has, and behave as though it's been denied when it hasn't.

**Comment by Nain57** [comment]
Hi and thank you for your feedback.

At code level, Accessibility permission granting leads to your device starting automatically a service inside Klick'r. When the application prompts you to enable Accessibility permission, it is because this internal service is not started.

So if your device says the permission is granted but Klick'r says it's not, it means the issue is on your device. I usually recommand to apply the guidelines available here https://dontkillmyapp.com, but unfortunately for you, it doesn't contains tips for your phone brand. 
But I know your issue can be reproduced on some Xiaomi devices, they kill the Klick'r service and forget to refresh the permission state. A work around is to disable the permision, grant it, disable it and grant it again a second time.


tldr; This is a known issue on multiple devices, the problem is in their falsy Android implementation and nothing can be done at application level.
